docs/sources/flow/reference/components/prometheus.operator.podmonitors.md

```diff
@@ -0,0 +1,193 @@
+---
+title: prometheus.operator.podmonitors
+labels:
+  stage: beta
+---
+
+# prometheus.operator.podmonitors
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" >}}
+
+`prometheus.operator.podmonitors` discovers [PodMonitor](https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.PodMonitor) resources in your kubernetes cluster and scrape the targets they reference. This component performs three main functions:
+
+1. Discover `PodMonitor` resources from your kubernetes cluster.
+2. Discover `Pods` in your cluster that match those `PodMonitors`.
+3. Scrape metrics from those `Pods`, and forward them to a receiver.
+
+The default configuration assumes the agent is running inside a kubernetes cluster, and uses the in-cluster config to access the kubernetes api. It can be run from outside the cluster by supplying connection info in the `client` block, but network level access to pods is required to scrape metrics from them.
```

Comment:
```suggestion
The default configuration assumes the agent is running inside a Kubernetes cluster, and uses the in-cluster config to access the Kubernetes API. It can be run from outside the cluster by supplying connection info in the `client` block, but network level access to pods is required to scrape metrics from them.
```